FirebaseInAppMessaging/Tests/Unit/FIRIAMClientInfoFetcherTests.m

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2020 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <OCMock/OCMock.h>
+#import <XCTest/XCTest.h>
+
+#import "FIRIAMClientInfoFetcher.h"
+
+#import <FirebaseInstallations/FIRInstallations.h>
+#import <FirebaseInstallations/FIRInstallationsAuthTokenResult.h>
+
+@interface FIRInstallationsAuthTokenResult (Tests)
+- (instancetype)initWithToken:(NSString *)token expirationDate:(NSDate *)expirationDate;
+@end
+
+@interface FIRIAMClientInfoFetcherTests : XCTestCase
+
+@property(nonatomic, strong) FIRIAMClientInfoFetcher *clientInfoFetcher;
+
+@end
+
+@implementation FIRIAMClientInfoFetcherTests
+
+- (NSDate *)future {
```

Comment:
Nit: If you don't need a specific time interval but juts a time in the future you may consider using [NSDate.distantFuture](https://developer.apple.com/documentation/foundation/nsdate/1415385-distantfuture?language=objc) property.